src/peergos/shared/user/app/StoreLocalAppData.java

```diff
@@ -9,11 +9,11 @@
  */
 public interface StoreLocalAppData {
 
-    CompletableFuture<List<String>> dirInternal(Path relativePath);
+    CompletableFuture<List<String>> dirInternal(Path relativePath, String username);
```

Comment:
This api doesn't make sense to me. Do we need a new ReadLocalAppData which this extends for secret links?

Reply:
I think I understand this now. But I need to think through the implications.